src/prefect/engine/state.py

```diff
@@ -107,6 +107,12 @@ def load_result(self, result: Result = None) -> "State":
             - State: the current state with a fully hydrated Result attached
         """
         if self.is_mapped():
+            self.map_states = [
+                s.load_result(result) if s is not None else None
+                for s in self.map_states  # type: ignore
+            ]
+            if self.map_states:
+                self.result = [getattr(s, "result", None) for s in self.map_states]
```

Comment:
If `self.map_states` is an empty list, should `self.result` also be an empty list (rather than `None`)?

Also, is the only time `s` won't have a `result` attribute if `s` is `None`? If so, I'd prefer a check for `None` rather than `getattr`, as it better indicates the logic of what's happening.

Reply:
> If self.map_states is an empty list, should self.result also be an empty list (rather than None)?

No effect either way; went with this approach to avoid changing tests

> is the only time s won't have a result attribute if s is None

yup that's right; so the check would look like:
```python
self.result = [s.result if s is not None else None for s in self.map_states]
```
if you think that's clearer then I can make the change

Reply:
Yeah, I think that's clearer. Otherwise looks good to me!

Reply:
Updated 👍